src/pytypes/onefuzztypes/models.py

```diff
@@ -32,6 +32,24 @@
 from .primitives import Container, PoolName, Region
 
 
+class EnumModel(BaseModel):
+    @root_validator(pre=True)
+    def exactly_one(cls: Any, values: Any) -> Any:
+        some = []
+
+        for field, val in values.items():
+            if val is not None:
+                some.append(field)
```

Comment:
no need to create and intermediate array you can just count the number of some(field) and validate the based on that number.

Reply:
True, but in this case I'm collecting the values on purpose so I can include them in the error message when validation fails. It's already helped me once!